src/storage/txn/store.rs

```diff
@@ -232,50 +220,54 @@ impl Store for FixtureStore {
             }
         });
 
-        let mut vec: Vec<_> = self
-            .data
-            .range((lower, upper))
-            .map(|(k, v)| {
-                let owned_k = k.clone();
-                let owned_v = if key_only {
-                    match v {
-                        Ok(_v) => Ok(vec![]),
-                        Err(e) => Err(e.maybe_clone().unwrap()),
-                    }
-                } else {
-                    match v {
-                        Ok(v) => Ok(v.clone()),
-                        Err(e) => Err(e.maybe_clone().unwrap()),
-                    }
-                };
-                (owned_k, owned_v)
-            })
-            .collect();
-
-        if desc {
-            vec.reverse();
-        }
+        let range_data_ref = self.data.clone();
+        // Erase the lifetime to be 'static.
+        let range_unsafe = unsafe { std::mem::transmute(range_data_ref.range((lower, upper))) };
 
         Ok(FixtureStoreScanner {
-            // TODO: Remove clone when GATs is available. See rust-lang/rfcs#1598.
-            data: vec.into_iter(),
+            key_only,
+            desc,
+            range_data_ref,
+            range_unsafe,
         })
     }
 }
 
 /// A Scanner that scans on fixtures.
+// TODO: Replace it using GATs when available, to avoid `Arc` cost.
+#[allow(clippy::type_complexity)]
 pub struct FixtureStoreScanner {
-    data: std::vec::IntoIter<(Key, Result<Vec<u8>>)>,
+    key_only: bool,
+    desc: bool,
+    /// Accessing this field is unsafe, since it's lifetime is faked. This field is valid only if
+    /// `range_data_ref` is valid. When `range_data_ref` is destroyed, this field is no longer
+    /// valid. Thus we must *NEVER* leak the reference of this field to the outside.
+    range_unsafe: btree_map::Range<'static, Key, Result<Vec<u8>>>,
+    #[allow(dead_code)]
+    range_data_ref: Arc<BTreeMap<Key, Result<Vec<u8>>>>,
```

Comment:
```suggestion
    _range_data_ref: Arc<BTreeMap<Key, Result<Vec<u8>>>>,
```

and remove the `#[allow(dead_code)]`.